Build on macOS fails to create delegator scripts for executables in sub-venvs

It tries to run the `chmod`:

    chmod: /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-evaluate: No such file or directory
    [...]

I am afraid that log output won't help much, nevertheless here it is:

    Successfully installed ocrd-segment-0.1.1
    make[1]: Nothing to be done for `ocrd-segment-from-masks'.
    make[1]: Nothing to be done for `ocrd-segment-from-coco'.
    make[1]: Nothing to be done for `ocrd-segment-extract-lines'.
    make[1]: Nothing to be done for `ocrd-segment-extract-regions'.
    make[1]: Nothing to be done for `ocrd-segment-extract-pages'.
    make[1]: Nothing to be done for `ocrd-segment-replace-original'.
    make[1]: Nothing to be done for `ocrd-segment-repair'.
    chmod +x /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-evaluate /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-from-masks /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-from-coco /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-extract-lines /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-extract-regions /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-extract-pages /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-replace-original /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-repair
    chmod: /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-evaluate: No such file or directory
    chmod: /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-from-masks: No such file or directory
    chmod: /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-from-coco: No such file or directory
    chmod: /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-extract-lines: No such file or directory
    chmod: /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-extract-regions: No such file or directory
    chmod: /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-extract-pages: No such file or directory
    chmod: /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-replace-original: No such file or directory
    chmod: /Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-repair: No such file or directory
    make: *** [/Users/stweil/src/github/OCR-D/venv-20200917/bin/ocrd-segment-evaluate] Error 1

--- a/Makefile
+++ b/Makefile
@@ -498,6 +498,8 @@ endef
 #  the sub-venv does not have the
 #  executable)
 # TODO: variant for relay to Docker
+ifeq ($(firstword $(subst ., ,$(MAKE_VERSION))),4)
+# make >=4 has file function
 define delegator
 #!/bin/bash
 . $(ACTIVATE_VENV) && $(BIN)/$(notdir $(1)) "$$@"
@@ -508,6 +510,18 @@ define delegate_venv
 $(foreach executable,$(1),$(file >$(executable),$(call delegator,$(executable))))
 chmod +x $(1)
 endef
+else
+# make <4 needs to echo (third recipe line must be empty!)
+define delegator
+@echo '#!/bin/bash' > $(1)
+@echo '. $(ACTIVATE_VENV) && $(BIN)/$(notdir $(1)) "$$@"' >> $(1)
+
+endef
+define delegate_venv
+$(foreach executable,$(1),$(call delegator,$(executable)))
+chmod +x $(1)
+endef
+endif
 
 # avoid making these .PHONY so they do not have to be repeated:
 # clstm tesserocr

